src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1344 @@ static ColliderCylinderInit D_80B37A5C = {
+        Matrix_MultVec3f(&colliderVec, &colliderPos);
+        this->colliderCylinder2.dim.pos.x = colliderPos.x;
+        this->colliderCylinder2.dim.pos.y = colliderPos.y;
+        this->colliderCylinder2.dim.pos.z = colliderPos.z;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35D90.s")
+    if ((this->fireTimer != 0) || ((this->actor.colorFilterTimer != 0) && (this->actor.colorFilterParams & 0x4000))) {
+        switch (limbIndex) {
+            case WOLFOS_LIMB_EYES:
+                bodyPartIndex = 0;
+                break;
+            case WOLFOS_LIMB_FRONT_RIGHT_LOWER_LEG:
+                bodyPartIndex = 1;
+                break;
+            case WOLFOS_LIMB_FRONT_LEFT_LOWER_LEG:
+                bodyPartIndex = 2;
+                break;
+            case WOLFOS_LIMB_THORAX:
+                bodyPartIndex = 3;
+                break;
+            case WOLFOS_LIMB_ABDOMEN:
+                bodyPartIndex = 4;
+                break;
+            case WOLFOS_LIMB_TAIL:
+                bodyPartIndex = 5;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_SHIN:
+                bodyPartIndex = 6;
+                break;
+            case 37: // There is no limb with index this large
+                bodyPartIndex = 7;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_PASTERN:
+                bodyPartIndex = 8;
+                break;
+            case WOLFOS_LIMB_BACK_LEFT_PAW:
+                bodyPartIndex = 9;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35E4C.s")
+        if (bodyPartIndex >= 0) {
+            Vec3f bodyPartPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35EE4.s")
+            Matrix_MultVec3f(&bodyPartVec, &bodyPartPos);
+            this->bodyPartsPos[bodyPartIndex].x = bodyPartPos.x;
+            this->bodyPartsPos[bodyPartIndex].y = bodyPartPos.y;
+            this->bodyPartsPos[bodyPartIndex].z = bodyPartPos.z;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B360E8.s")
+static void* sWolfosEyeTextures[] = { gWolfosNormalEyeOpenTex, gWolfosNormalEyeHalfTex, gWolfosNormalEyeNarrowTex,
+                                      gWolfosNormalEyeHalfTex };
+static void* sWhiteWolfosEyeTextures[] = { gWolfosWhiteEyeOpenTex, gWolfosWhiteEyeHalfTex, gWolfosWhiteEyeNarrowTex,
+                                           gWolfosWhiteEyeHalfTex };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B361A0.s")
+void EnWf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnWf* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36288.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_wf.c", 2157);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B36328.s")
+    if ((this->action != WOLFOS_ACTION_WAIT_TO_APPEAR) || !this->isInvisible) {
```

Comment:
```suggestion
    if (!((this->action == WOLFOS_ACTION_WAIT_TO_APPEAR) && this->isInvisible)) {
```

Reply:
kinda feel like the way it was before is easier to wrap your head around, at least for me

Reply:
I got the variable the wrong way round the first time when it was the old way, so I'm not convinced of either way tbh.

Reply:
what was the wrong way around? sorry, confused

Reply:
ah ok
well either way, i still think it makes sense as an or. this condition is preventing the wolfos from drawing under two different conditions, either if its invisible or its waiting to appear. anding them and negating the result treats it as one condition in a way that is confusing for me

Reply:
I'm reading this as "if isn't (invisible and waiting to appear)" which btw makes it look like the invisible struct member may be able to get a better name

Reply:
Maybe invisible -> didntAppearYet idk

Reply:
I was more than sure there was an invisible type of wolfos, which is why to me they should have been treated separate. but it seems that isnt the case

I dont even follow what this variable is doing, it seems like its stopping drawing at random points in its execution, like when it gets damaged and stuff. But ive never seen them just disappear temporarily lol

Reply:
Yeah, I named it based on what it does in the draw... don't really understand what the rest of the code is doing with it. Maybe I should get the breakpoints out.

Reply:
`isInvisible` looks like `hasAppeared`

since it's unused unless `action == WOLFOS_ACTION_WAIT_TO_APPEAR` I think whoever wrote this actor got confused at some point

Reply:
Mm, not sure what to do about this then. It seems redundant unless they intended there to be invisible Wolfos (although if they did, wouldn't they treat them like invisible Keese?)

Reply:
keese is an exceptional case, generally invisible vs visible versions of an actor are handled inside the same draw function.
and fwiw geldb has this exact same conditional and has it as or right now. though in that case i also dont understand why it becomes "invisble" when it rolls forward and backward and such lol

Reply:
Okay, I changed it back to be the same as EnGeldB, since they should probably be consistent whatever we decide to do.